src/utils/PluginDriver.ts

```diff
@@ -360,3 +392,36 @@ export class PluginDriver {
 		}
 	}
 }
+
+// provides uniqueness for unfulfilled hook actions and the dispatch order is useful for debugging
+let actionCount = 0;
+
+function formatAction(pluginName: string, hookName: string, args: Parameters<any>): string {
+	let action = `${++actionCount}: (${pluginName}) ${hookName}`;
+	const s = JSON.stringify;
+	switch (hookName) {
+		case 'resolveId':
+			action += ` ${s(args[0])} ${s(args[1])}`;
+			break;
+		case 'load':
+			action += ` ${s(args[0])}`;
+			break;
+		case 'transform':
+			action += ` ${s(args[1])}`;
+			break;
+	}
+	return action;
+}
+
+if (typeof process === 'object' && process && typeof process.on === 'function') {
```

Comment:
I am not a fan of placing node-specific logic here, even if well guarded, but I know you tend to disagree. Still, as this will not work in the browser anyway, why not do something similar to what we do for fs and path: Put the logic into a separate file that is replaced with a no-op browser version for the browser build?
Then you do not need the checks, and we could even move out more logic. E.g. the file could export two functions "addUnresolvedAction(actionTuple)" and "resolveAction(actionTuple)" and everything else including the reporting would be a black box. Very likely, tree-shaking would then even remove the calls to these functions if they do not do anything in the browser version.